server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4591,6 +4591,10 @@ public void finalizeStop(VirtualMachineProfile profile, Answer answer) {
             throw new PermissionDeniedException("The owner of " + vm + " is disabled: " + vm.getAccountId());
         }
 
+        if (vm.getState()== State.Running){
+            throw new InvalidParameterValueException("The virtual machine "+ vm.getDisplayName()+ " is already running");
```

Comment:
Do we want to return ths displayname here and not the regular name of the VM?

Reply:
or even the id? Who is this message for, I think it can be directed both at an end user and at an admin. So maybe regular name or id/displayname combination?